runtime/vm/visible.c

```diff
@@ -29,6 +29,62 @@
 #include "util_api.h"
 #include "vm_internal.h"
 
+
+
+/**
+ * Check module access from srcModule to destModule
+ *
+ * @param[in] currentThread the current J9VMThread
+ * @param[in] vm the javaVM
+ * @param[in] srcRomClass the accessing class
+ * @param[in] srcModule the module of the src class
+ * @param[in] destRomClass the accessing class
+ * @param[in] destModule the module of the dest class
+ * @param[in] destPackageID packageID of the dest class
+ * @param[in] lookupOptions J9_LOOK* options
+ *
+ * @return 	J9_VISIBILITY_ALLOWED if the access is allowed,
+ * 			J9_VISIBILITY_MODULE_READ_ACCESS_ERROR if module read access error occurred,
+ * 			J9_VISIBILITY_MODULE_PACKAGE_EXPORT_ERROR if module package access error
+ */
+VMINLINE IDATA
+checkModuleAccess (J9VMThread *currentThread, J9JavaVM* vm, J9ROMClass* srcRomClass, J9Module* srcModule, J9ROMClass* destRomClass, J9Module* destModule, UDATA destPackageID, UDATA lookupOptions)
+{
+	UDATA result = J9_VISIBILITY_ALLOWED;
+	if (srcModule != destModule) {
+		UDATA rc = ERRCODE_GENERAL_FAILURE;
+		if (!J9_ARE_ALL_BITS_SET(lookupOptions, J9_LOOK_REFLECT_CALL)) {
+			if (!isAllowedReadAccessToModule(currentThread, srcModule, destModule, &rc)) {
+				Trc_VM_checkVisibility_failed_with_errortypeRomClass(currentThread,
+						srcRomClass, J9UTF8_LENGTH(J9ROMCLASS_CLASSNAME(srcRomClass)), J9UTF8_DATA(J9ROMCLASS_CLASSNAME(srcRomClass)), srcModule,
+						destRomClass, J9UTF8_LENGTH(J9ROMCLASS_CLASSNAME(destRomClass)), J9UTF8_DATA(J9ROMCLASS_CLASSNAME(destRomClass)), destModule, "read access not allowed");
+				result = J9_VISIBILITY_MODULE_READ_ACCESS_ERROR;
+			}
+		}
+
+		if (J9_VISIBILITY_ALLOWED == result) {
+			const U_8* packageName = NULL;
+			UDATA packageNameLength = 0;
+			J9PackageIDTableEntry entry = {0};
+
+			entry.taggedROMClass = destPackageID;
+			packageName = getPackageName(&entry, &packageNameLength);
+
+			omrthread_monitor_enter(vm->classLoaderModuleAndLocationMutex);
+			if (!isPackageExportedToModuleWithName(currentThread, destModule, (U_8*) packageName, (U_16) packageNameLength, srcModule, J9_IS_J9MODULE_UNNAMED(vm, srcModule), &rc)) {
+				Trc_VM_checkVisibility_failed_with_errortypeRomClass(currentThread,
```

Comment:
Should this tracepoint include the `rc` value?  Otherwise, it should be named something like `ignored`.

Reply:
by `RC` do you mean `result`?

`result` is not passed in to the tracepoint